core/src/main/java/dk/alexandra/fresco/lib/lp/LPSolver.java

```diff
@@ -281,37 +297,50 @@ public LPOutput(LPTableau tableau, Matrix<DRes<SInt>> updateMatrix, List<DRes<SI
     }
   }
 
-  protected class LpState implements DRes<LpState> {
+  protected static class LpState {
 
-    public DRes<BigInteger> terminationOut;
-    private LPTableau tableau;
-    private Matrix<DRes<SInt>> updateMatrix;
-    public List<DRes<SInt>> enteringIndex;
-    public DRes<SInt> pivot;
-    public List<BigInteger> enumeratedVariables;
-    public List<DRes<SInt>> basis;
-    public DRes<SInt> prevPivot;
+    private final int iteration;
+    private final boolean terminated;
+    private final LPTableau tableau;
+    private final Matrix<DRes<SInt>> updateMatrix;
+    private final DRes<SInt> pivot;
+    private final List<BigInteger> enumeratedVariables;
+    private final List<DRes<SInt>> basis;
+    private final DRes<SInt> prevPivot;
 
-    public LpState(BigInteger terminationOut, LPTableau tableau, Matrix<DRes<SInt>> updateMatrix,
-        List<DRes<SInt>> enteringIndex, DRes<SInt> pivot, List<BigInteger> enumeratedVariables,
-        List<DRes<SInt>> basis, DRes<SInt> prevPivot) {
-      this.terminationOut = () -> terminationOut;
+    public LpState(List<DRes<SInt>> basis, Matrix<DRes<SInt>> updateMatrix,
```

Comment:
The constructors of the LpState are pretty complex. We should at least sort the arguments in the same order in each constructor.